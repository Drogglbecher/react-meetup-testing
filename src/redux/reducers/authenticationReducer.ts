// Actions
import {
  Action,
  Failed,
  Logout,
  SubmitCredentialsSuccess,
} from '../actions/authenticationActions';

export type State = {
  authenticated: boolean;
  username: string;
  error: string;
};

export const initialState = {
  authenticated: false,
  username: '',
  error: '',
};

export default function(state: State = initialState, action: Action) {
  switch (action.type) {
    case SubmitCredentialsSuccess: {
      return {
        authenticated: true,
        error: '',
        username: action.payload,
      };
    }

    case Logout: {
      return {
        authenticated: false,
        error: '',
        username: '',
      };
    }

    case Failed: {
      return {
        authenticated: false,
        error: action.payload,
        username: '',
      };
    }

    default:
      return state;
  }
}
