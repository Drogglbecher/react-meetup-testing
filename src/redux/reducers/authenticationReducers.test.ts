// Actions
import {
  failed,
  logout,
  submitCredentialsSuccess,
} from '../actions/authenticationActions';

// Reducers
import authenticationReducer, {initialState} from './authenticationReducer';

describe('Authentication Reducers', () => {
  it('should set initial state', () => {
    const action = {type: 'dummy_action', payload: null};
    expect(authenticationReducer(undefined, action)).toMatchSnapshot();
  });

  it('should set `user` and `authenticated` to true when a `SubmitCredentialsSuccess` action is dispatched', () => {
    const action = submitCredentialsSuccess('user');
    expect(authenticationReducer(undefined, action)).toMatchSnapshot();
  });

  it('should set `error` and `user` to null when a `Failed` action is dispatched', () => {
    const initialStateLocal = {
      ...initialState,
      user: 'user',
    };

    const action = failed('Nix gut.');
    expect(authenticationReducer(initialStateLocal, action)).toMatchSnapshot();
  });

  it('should set `authenticated` to false when a `Logout` action is dispatched', () => {
    const initialStateLocal = {
      ...initialState,
      authenticated: true,
      user: 'user',
    };

    const action = logout();
    expect(authenticationReducer(initialStateLocal, action)).toMatchSnapshot();
  });
});
