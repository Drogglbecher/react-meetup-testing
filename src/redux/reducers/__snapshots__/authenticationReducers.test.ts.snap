// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication Reducers should set \`authenticated\` to false when a \`Logout\` action is dispatched 1`] = `
Object {
  "authenticated": false,
  "error": "",
  "username": "",
}
`;

exports[`Authentication Reducers should set \`error\` and \`user\` to null when a \`Failed\` action is dispatched 1`] = `
Object {
  "authenticated": false,
  "error": "Nix gut.",
  "username": "",
}
`;

exports[`Authentication Reducers should set \`user\` and \`authenticated\` to true when a \`SubmitCredentialsSuccess\` action is dispatched 1`] = `
Object {
  "authenticated": true,
  "error": "",
  "username": "user",
}
`;

exports[`Authentication Reducers should set initial state 1`] = `
Object {
  "authenticated": false,
  "error": "",
  "username": "",
}
`;
